nano/node/node.cpp

```diff
@@ -3280,6 +3281,24 @@ void nano::active_transactions::adjust_difficulty (nano::block_hash const & hash
 	}
 }
 
+void nano::active_transactions::update_active_difficulty ()
+{
+	assert (!mutex.try_lock ());
+	uint64_t difficulty (node.network_params.publish_threshold);
+	if (!roots.empty ())
+	{
+		difficulty = roots.get<1> ().find ((roots.get<1> ().begin ()->adjusted_difficulty + (--roots.get<1> ().end ())->adjusted_difficulty) / 2)->adjusted_difficulty;
+	}
+	difficulty_cb.push_front (difficulty);
+	uint128_t sum (0);
+	for (auto & i : difficulty_cb)
+	{
+		sum += i;
```

Comment:
This can be replaced with the `std::accumulate` algorithm

Reply:
Does that work with boost's uint128_t? I've tried that in the past and failed. Very nice if there's a way.

Reply:
`auto sum = std::accumulate (difficulty_cb.begin (), difficulty_cb.end (), uint128_t (0));` should work. Perhaps you just used a plain 0, std::accumulate does it's automatic type deduction based on the `init` variable.

Reply:
The init type could well've been wrong. Cool.